CIP-XXXX/README.md

```diff
@@ -0,0 +1,173 @@
+}
+```
+
+### Aditional Error Types
+
+#### CompletedTxError
+
+```ts
+CompletedTxErrorCode = {
+	NotFound: 1,
+	NotReady: 2
+}
+```
+
+* NotFound - The transaction with the given id was not found.
+* NotReady - The transaction with the given id is not ready yet. 
+
+## Motivation: why is this CIP necessary?
+In order to facilitate future dApp development, we will need a way for dApps to communicate with multisig wallets. Given the unique complexities of native script based addresses, special provisions need to be made to make the connector compatible with them.
+
+Specifically, apps building transactions need to be able to get the following information from the wallet:
+
+- Script descriptor
+- Script Requirements list
+- Collateral donator (since native script based addresses cannot provide collateral for transactions)
+
+Additionally, apps need to be able to submit a transaction to the wallet for signing in an asynchronous manner, as gathering of signatures can take a long time and each wallet provider will have its own way of handling this process.
+
+Finally, the signTx() and signData() endpoints will have to be disabled when using this extension since they are not compatible with native script based addresses.
+
+## Rationale for the required data
+- Script descriptor: Any transaction consuming a UTxO from a native script based address must attach the corresponding script.
+- Script Requirements list: 
+	-- dApps need to know the number of signers to calculate the fees correctly. 
+	-- Including the correct "required signers" brings UX improvements. 
+	-- dApps need to know if the script has any before-after requirements to set the correct validBefore and TTL values.
+- Collateral donator Native script based addresses cannot provide collateral for Plutus transactions.
+
+### Additional API Endpoints
+
+#### api.getCollateralAddress(): Promise\<Address>
+
+For Plutus V2 and later, partial collateral is supported. This function returns an address that can be used to add collateral to a transaction. The address returned must be owned by one of the signers in the list of signers returned by `api.getScriptRequirements()`. 
+
+dApp developers can choose to use this address to add collateral to a transaction, or they can choose to use the `api.getCollateral()` function to get a list of UTXOs that can be used as collateral. If the dApp chooses to use this address, they must ensure that the address is not used for any other purpose, as the wallet may be using it to track collateral, and that the collateral return address is the same one.
+
+#### api.getScriptRequirements: Promise\<ScriptRequirement[]>
+
+Errors: `APIError`
+
+Returns a list of ScriptRequirements that will be used to validate any transaction sent to the wallet.
+
+#### api.getScript(): Promise\<cbor\<nativeScript>>
+
+Errors: `APIError`
+
+Returns the CBOR-encoded native script that controls this wallet. 
+
+#### api.submitUnsignedTx(tx: cbor\<unsignedTransaction>): Promise\<hash32>
```

Comment:
Maybe we could find a way to utilize existing sign() and submit() instead. For example: to address sign() being a long-running operation, wallets could make `sign()` idempotent and associate subsequent calls with an already ongoing signing operation. Wallets have to maintain state of transaction being signed anyway, so it shouldn't be too difficult to implement. This way, dapps could recover the state (get a callback when signing is complete) when they're reloaded.

Reply:
This is a complex problem with many possible solutions.

Long polling, callbacks, web-hooks. I recommend this be addressed in other CIPs each of witch will implement a single solution.  That will save us from getting locked into a specific mode of operation at this stage.